src/LanguageServer/Protocol/Handler/CodeLens/CodeLensResolveHandler.cs

```diff
@@ -48,11 +49,13 @@ public async Task<LSP.CodeLens> HandleRequestAsync(LSP.CodeLens request, Request
             ]
         };
 
-        // If the request is for an older version of the document, return a request with '- references'
+        // If the request is for an older version of the document, throw an exception so the client knows to re-query us.
         if (resolveData.SyntaxVersion != currentDocumentSyntaxVersion.ToString())
         {
-            context.TraceInformation($"Requested syntax version {resolveData.SyntaxVersion} does not match current version {currentDocumentSyntaxVersion}");
-            return request;
+            throw new LocalRpcException($"Resolve version {resolveData.SyntaxVersion} does not match current version {currentDocumentSyntaxVersion}")
```

Comment:
Placeholders should be wrapped in single quotes in exception messages

```suggestion
            throw new LocalRpcException($"Resolve version '{resolveData.SyntaxVersion}' does not match current version '{currentDocumentSyntaxVersion}'")
```